### DISCUSSION TITLE
Connect with PnPOnline

### DISCUSSION THREAD

**Discussion by MattK-CPA** [pnp/powershell#144] [Q&A] [answered]
I am hoping that someone will be willing to assist me with some advice.
I installed the latest version of the **PnP.PowerShell** module Version 1.1.0. 
I have done the `Register-PnPManagementShellAccess` and in the Azure Active Directory Enterprise applications list I can see the ‘PnP Management Shell’ so I assume that requirement has been achieved successfully.
However, when I attempt to use 
`Connect-PnPOnline -Url https://mytenant.sharepoint.com`
I get prompted for credentials, I enter the username and password details and then, I continue to get an error
`Connect-PnPOnline: Federated service at https://autologon.microsoftazuread-sso.com/... returned error: Authentication Failure`
I am attempting to login/sign-in interactively with an account that has MFA requirements. I am wondering whether that may have something to do with the problems and errors.  Previously I thought that I was able to use an app password that was associated to the account to successfully connect.
If I use 
`Connect-PnPOnline -Url https://tenant.sharepoint.com -UseWebLogin`
That will work satisfactorily, and I can run basic Cmdlets eg. get-PnPList.  However, as recorded elsewhere in the documentation, this UseWebLogin connection method has limitations and will not be able to acquire an access token for Microsoft Graph and make calls there behind the scenes. 

I really want to be able to use the PnP.PowerShell module for the functionality of the provisioning engine, eg. `Get-PnPSiteTemplate`, so it would be useful to find a way to connect using account credentials and have this work.
Can anyone make any suggestions about why I am getting the “Authentication Failure” error when I use username and password credentials and what I may need to do to avoid the error? Are there any other details I am missing with regards to using Connect-PnPOnline in the new PnP.PowerShell module ?
Thanks.

**Comment by erwinvanhunen** [comment, marked as answer]
Hi Matt,

We're working on an interactive login, supporting MFA, that allows us to acquire tokens. This is a similar login as using credentials and using the PnP Management Shell (we will also allow you provide your own azure app id). It will very likely show up in one of the upcoming nightly builds. Until that time you're limited to -UseWebLogin with its limitations unfortunately.

/Erwin

**Comment by erwinvanhunen** [comment]
Update: it will be available in the next nightly release: 1.2.1-nightly. You can upgrade to that version when available with

```Update-Module -Name PnP.PowerShell -AllowPrerelease```

After installation that version you can connect as follows:

```Connect-PnPOnline -Url [yoururl] -Interactive```
